extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLSubscriptionHandler.java

```diff
@@ -0,0 +1,118 @@
+    public SmallRyeGraphQLSubscriptionHandler(Config config, CurrentIdentityAssociation currentIdentityAssociation,
+            CurrentVertxRequest currentVertxRequest) {
+        super(currentIdentityAssociation, currentVertxRequest);
+        this.executionErrorsService = new ExecutionErrorsService(config);
+    }
+
+    @Override
+    protected void doHandle(final RoutingContext ctx) {
+        if (ctx.request().headers().contains(HttpHeaders.UPGRADE, HttpHeaders.WEBSOCKET, true)) {
+            ctx.request().toWebSocket(new Handler<AsyncResult<ServerWebSocket>>() {
+                @Override
+                public void handle(AsyncResult<ServerWebSocket> event) {
+                    if (event.succeeded()) {
+                        ServerWebSocket socket = event.result();
+
+                        socket.textMessageHandler(new Handler<String>() {
+                            @Override
+                            public void handle(String message) {
+                                final AtomicReference<Subscription> subscriptionRef = new AtomicReference<>();
+                                JsonObject jsonInput = inputToJsonObject(message);
+                                ExecutionResponse executionResponse = getExecutionService()
+                                        .execute(jsonInput);
+
+                                ExecutionResult executionResult = executionResponse.getExecutionResult();
+                                if (executionResult != null) {
+
+                                    // If there is error on the query, we can not start a subscription
+                                    if (executionResult.getErrors() != null && !executionResult.getErrors().isEmpty()) {
+                                        socket.writeTextMessage(executionResponse.getExecutionResultAsString());
+                                        socket.close();
+                                    } else {
+                                        Publisher<ExecutionResult> stream = executionResponse.getExecutionResult().getData();
+
+                                        if (stream != null) {
+                                            stream.subscribe(new Subscriber<ExecutionResult>() {
+
+                                                @Override
+                                                public void onSubscribe(Subscription s) {
+                                                    subscriptionRef.set(s);
+                                                    s.request(1);
+                                                }
+
+                                                @Override
+                                                public void onNext(ExecutionResult er) {
+                                                    Object response = er.getData();
+                                                    if (!socket.isClosed()) {
+                                                        socket.writeTextMessage(JSONB.toJson(response));
```

Comment:
Maybe the output format is specified somewhere, but I don't like that we only return the data itself, but leave out execution errors. Shouldn't we send the whole `er` object, including errors?

Reply:
Yes we should.... I did not think about that .... ok let me look at that....

Reply:
Damn, this is big mistake. Same error in SmallRye Servlet module....

Reply:
@jmartisk  - Ok this is now fixed. Missed one line. This is also an issue in SmallRye Servlet module (not used here) that I think is used in Wildfly, so we would need a new release for WF. Thanks for catching this :)